src/create.rs

```diff
@@ -464,47 +540,46 @@ pub enum Error {
 #[cfg(test)]
 mod test {
     use super::*;
+    use bitcoin::TxIn;
 
     #[test]
     fn channel_creation() {
```

Comment:
Moving that to Rust doc would allow an easy documentation of the crate.

Reply:
Is documentation executed though? I thought it was just compilation.

Reply:
Documentation is executed.
You are thinking of examples which are only compiled.

Reply:
I prefer the tests being _real tests_ for the time being. Once code evolves to a library the Rust doc approach might be more useful.

Reply:
> I prefer the tests being real tests for the time being. Once the code evolved to a library the Rust doc approach might be more useful.

Rust doc tests are not `fake tests`. They are tests that are part of the documentation.
Can you please explain a bit more the reason behind your preference? Or is just because we are not familiar with writing embedded code in rust doc?

Reply:
Rust doc is meant for documenting a rust project and are meant to be used as examples (I think we are on the same page here?). 
The tests we are writing atm are used to show that our code works. 

What I want to say is that one does not exclude the other but the format should be different.